modules/core/src/main/java/org/apache/ignite/internal/processors/job/GridJobProcessor.java

```diff
@@ -2282,4 +2311,49 @@ private JobsMap(int initCap, float loadFactor, int concurLvl) {
             return sizex();
         }
     }
+
+    /**
+     * Callback on changing task attributes.
+     *
+     * @param sesId Session ID.
+     * @param jobId Job ID.
+     * @param attrs Changed attributes.
+     */
+    public void onChangeTaskAttributes(IgniteUuid sesId, IgniteUuid jobId, Map<?, ?> attrs) {
```

Comment:
Why is it public?

Reply:
We need to call from class `GridTaskProcessor#sendSessionAttributes` and `GridJobProcessor#processTaskSessionRequest` and therefore `public`.